### DISCUSSION TITLE
Integrate angular components

### DISCUSSION THREAD

**Discussion by florianbaillagou** [GrapesJS/grapesjs#4835] [Q&A]
Hello everyone,

I'm currently playing with GrapesJS and happily surprised with the result.

So far i could create my own block like an Openlayers one and it's working !

Now, i'm trying to create blocks for angular components but this is where i'm a bit stuck. I'm not able to render the component.

After checking for discussions, I could see some Vuejs and React integration but it's a bit hard a understand as i'm not really confortable with all the frameworks.

Before losing to much time, can you confirm it's possible to integrate angular component into GrapesJS ? If yes, do you think i can extend the work done on Vuejs/React integration ?

Best regards,

**Comment by zorkind** [comment]
Hey this might interest you.

https://github.com/rakutentech/grapesjs-angular

🤔

**Reply by florianbaillagou** [reply]
Hey !

I could understand that this is not a library to inject Angular components into the editor, but a wrapper to provide the editor into an Angular component.

But the solution might be with Angular Dynamic Component Loader. If i'm not mistaken, this is what is done on Vuejs/React integration but i'm a bit lost on how to translate Vuejs/React work to Angular. I'll try to dig this part and keep you updated.

**Comment by florianbaillagou** [comment]
Evening,

I finally did what i wanted ! It was with a simple component but it's working fine so far.

I used this article to perform what i wanted to do : https://www.htmlelements.com/docs/angular-dynamic-component-loading/

I will try with a more complex one in the next following days and also check about restoring the editor from storage.
